great_expectations/datasource/fluent/interfaces.py

```diff
@@ -383,7 +384,7 @@ class Datasource(
     assets: MutableMapping[str, _DataAssetT] = {}
 
     # private attrs
-    _data_context = pydantic.PrivateAttr()
+    _data_context: GXDataContext = pydantic.PrivateAttr()
```

Comment:
Why rename this type?

Reply:
@NathanFarmer The original import came from `interface.py` where line 387 contains `_data_context: GXDataContext = pydantic.PrivateAttr()`.  It is then used in #7425 as well.  Thank you.